test/test_uri_parser.py

```diff
@@ -553,6 +554,119 @@ def test_port_with_whitespace(self):
         with self.assertRaisesRegex(ValueError, r"Port contains whitespace character: '\\n'"):
             parse_uri("mongodb://localhost:27\n017")
 
+    # Initial DNS Seedlist Discovery prose tests
+    # https://github.com/mongodb/specifications/blob/0a7a8b5/source/initial-dns-seedlist-discovery/tests/README.md#prose-tests
+
+    def test_1_allow_srv_hosts_with_fewer_than_three_dot_separated_parts(self):
+        with patch("dns.resolver.resolve"):
+            parse_uri("mongodb+srv://localhost/")
+            parse_uri("mongodb+srv://mongo.local/")
+
+    def test_2_throw_when_return_address_does_not_end_with_srv_domain(self):
+        test_cases = [
+            {
+                "query": "_mongodb._tcp.localhost",
+                "mock_target": "localhost.mongodb",
+                "expected_error": "Invalid SRV host",
+            },
+            {
+                "query": "_mongodb._tcp.blogs.mongodb.com",
+                "mock_target": "blogs.evil.com",
+                "expected_error": "Invalid SRV host",
+            },
+            {
+                "query": "_mongodb._tcp.blogs.mongo.local",
+                "mock_target": "test_1.evil.com",
+                "expected_error": "Invalid SRV host",
+            },
+        ]
+        for case in test_cases:
```

Comment:
This for loop looks to be identical between these 3 tests. Can you refactor them to reduce the code duplication?

Reply:
Done in 5c823fcd4037e68a8b46b286f9fc0e7f030c483e